sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/AbstractTypesShouldNotHaveConstructorsTests.cs

```diff
@@ -34,5 +34,14 @@ public void NoPublicConstructor()
         {
             Verifier.VerifyAnalyzer(@"TestCases\AbstractTypesShouldNotHaveConstructors.cs", new AbstractTypesShouldNotHaveConstructors());
         }
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void NoPublicConstructor_TopLevelStatements()
+        {
+            Verifier.VerifyAnalyzer(@"TestCases\AbstractTypesShouldNotHaveConstructors.TopLevelStatements.cs", new AbstractTypesShouldNotHaveConstructors(),
+                options: ParseOptionsHelper.FromCSharp9, checkMode: CompilationErrorBehavior.Ignore);
```

Comment:
there's a compilation error - the compiler says it must be inside an executable project